src/App.tsx

```diff
@@ -1,27 +1,61 @@
 import React from 'react';
-import { useSelector } from 'react-redux';
-
 import './App.scss';
-import Start from './components/Start';
-import { Finish } from './components/Finish';
-
-import { isLoading, getMessage } from './store';
+import Todos from './components/Todos';
+import { connect, useSelector } from 'react-redux';
+import { getIsLoaded } from './store'
+import { CustomTodo, RootState } from './types';
+import { getCustomTodos } from './api';
 
+const App = ({
+  buttonText,
+  setCustomTodos,
+  setButtonText,
+  setIsLoaded
+}: {
+  setCustomTodos: (customTodos: CustomTodo[]) => void;
+  setIsLoaded: (isLoaded: boolean) => void;
+  setButtonText: (buttonText: string) => void;
+  buttonText: string;
+}) => {
 
-const App = () => {
-  const loading = useSelector(isLoading);
-  const message = useSelector(getMessage) || 'Ready!';
+  const handleButtonClick = () => {
+    setButtonText('Data is loaded now...')
+    getCustomTodos().then(customTodos => {
+      setCustomTodos(customTodos);
+      setIsLoaded(true);
+    });
+  }
 
   return (
     <div className="App">
       <h1>Redux list of todos</h1>
-      <h2>{loading ? 'Loading...' : message}</h2>
-
-      <Start title="Start loading" />
-      <Finish title="Succeed loading" message="Loaded successfully!" />
-      <Finish title="Fail loading" message="An error occurred when loading data." />
+      {useSelector(getIsLoaded)
+        ? <Todos />
+        : <button
+          onClick={handleButtonClick}
+        >{buttonText}</button>}
```

Comment:
```suggestion
      {
        useSelector(getIsLoaded)
          ? <Todos />
          : (
            <button
              onClick={handleButtonClick}
            >
              {buttonText}
            </button>
          )
      }
```
this is how your formating should look like